### PR TITLE
RESTWS-649 - Correct host property in OpenAPI Spec

### PR DESCRIPTION
**Pull request by gayanW**
* Make host and basePath not mandatory in SwaggerSpecificationCreator
* Set host and basePath to correct values in  SwaggerSpecificationController

Issue: https://issues.openmrs.org/browse/RESTWS-649

### CODE REVIEW

File: omod-common/src/main/java/org/openmrs/module/webservices/docs/swagger/SwaggerSpecificationCreator.java [hunk 1]
```diff
@@ -806,38 +818,6 @@ private void addSubclassOperations() {
 		}
 	}
 	
-	@Deprecated
-	private List<org.openmrs.module.webservices.docs.swagger.Parameter> getParametersListForSearchHandlers(
-	        String resourceName, String searchHandlerId, int queryIndex) {
-		List<org.openmrs.module.webservices.docs.swagger.Parameter> parameters = new ArrayList<org.openmrs.module.webservices.docs.swagger.Parameter>();
-		String resourceURL = getResourceUrl(getBaseUrl(), resourceName);
-		for (SearchHandlerDoc searchDoc : searchHandlerDocs) {
```
Comment (dkayiwa) [inline omod-common/src/main/java/org/openmrs/module/webservices/docs/swagger/SwaggerSpecificationCreator.java]: Is removing this block of code related to? "Correct host property in OpenAPI Spec"

File: omod-common/src/main/java/org/openmrs/module/webservices/docs/swagger/SwaggerSpecificationCreator.java [hunk 2]
```diff
@@ -87,10 +86,27 @@
 	
 	private Logger log = Logger.getLogger(this.getClass());
 	
-	public SwaggerSpecificationCreator(String baseUrl) {
-		this.baseUrl = baseUrl;
```
Comment (dkayiwa) [inline omod-common/src/main/java/org/openmrs/module/webservices/docs/swagger/SwaggerSpecificationCreator.java]: Is removing the lines below related to? "Correct host property in OpenAPI Spec"

**Comment by dkayiwa** [comment]
@gayanW this pull request seems to have more changes that needed for just "Correct host property in OpenAPI Spec". Can you create a separate pull request for other things instead of including them all in this?

File: omod-common/src/main/java/org/openmrs/module/webservices/docs/swagger/SwaggerSpecificationCreator.java (hunk 2 above)
Comment (gayanW) [reply omod-common/src/main/java/org/openmrs/module/webservices/docs/swagger/SwaggerSpecificationCreator.java]: We no longer use the baseUrl constructor. However the `searchHandlerDocs` instance variable requires baseUrl param to initiate. `searchHandlerDocs` is only referenced by getParametersListForSearchHandlers() which is unused. So it makes sense to remove these here.

File: omod-common/src/main/java/org/openmrs/module/webservices/docs/swagger/SwaggerSpecificationCreator.java (hunk 1 above)
Comment (gayanW) [reply omod-common/src/main/java/org/openmrs/module/webservices/docs/swagger/SwaggerSpecificationCreator.java]: See the comment above.

File: omod-common/src/main/java/org/openmrs/module/webservices/docs/swagger/SwaggerSpecificationCreator.java (hunk 2 above)
Comment (dkayiwa) [reply omod-common/src/main/java/org/openmrs/module/webservices/docs/swagger/SwaggerSpecificationCreator.java]: I do not refute the fact that we no longer need some code. What am simply saying is that if removing it is not related to "Correct host property in OpenAPI Spec", then it should be in a separate pull request.

**Comment by gayanW** [comment]
Please see now. baseUrl was being used to generate the host, basePath, and scheme. Yet the logic there was invalid. So it needs to be removed.

Now in the controller we pass the accurate values.

**Comment by dkayiwa** [comment]
@gayanW do you mind squashing the commits?

**Comment by gayanW** [comment]
Done.

File: omod/src/main/java/org/openmrs/module/webservices/rest/web/controller/SwaggerSpecificationController.java [hunk 3]
```diff
@@ -27,45 +27,12 @@
...
 	String getSwaggerSpecification(HttpServletRequest request) throws Exception {
 		
-		String swaggerSpecificationJSON = "";
-		StringBuilder baseUrl = new StringBuilder();
-		String scheme = request.getScheme();
-		int port = request.getServerPort();
-		
-		baseUrl.append(scheme); // http, https
-		baseUrl.append("://");
-		baseUrl.append(request.getServerName());
```
Comment (dkayiwa) [inline omod/src/main/java/org/openmrs/module/webservices/rest/web/controller/SwaggerSpecificationController.java]: Does this ticket involve removing all these lines of code (from line 30 - 68)?
Are you sure we are not having any regressions here?

File: omod/src/main/java/org/openmrs/module/webservices/rest/web/controller/SwaggerSpecificationController.java [hunk 4]
```diff
@@ -27,45 +27,12 @@
...
-		String urlWithoutScheme = "";
-		
-		/* Swagger appends scheme to urls, so we should remove it */
-		if (scheme.equals("http"))
-			urlWithoutScheme = resourcesUrl.replace("http://", "");
-		
-		else if (scheme.equals("https"))
-			urlWithoutScheme = resourcesUrl.replace("https://", "");
-		
-		SwaggerSpecificationCreator creator = new SwaggerSpecificationCreator(urlWithoutScheme);
```
Comment (gayanW) [inline omod/src/main/java/org/openmrs/module/webservices/rest/web/controller/SwaggerSpecificationController.java]: All this logic does nothing but build the baseUrl (urlWithoutScheme) string and then pass it to the SwaggerSpecificationCreator(..). As noted before generated string was invalid. Thus results in invalid host, and basePath in the spec. No there's no regression.

**Pull request merged**